services/sinner/api/server.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/ipfs/go-log"
+	"github.com/soheilhy/cmux"
+	"github.com/synapsecns/sanguine/core"
+	"github.com/synapsecns/sanguine/core/ginhelper"
+	"github.com/synapsecns/sanguine/core/metrics"
+	serverConfig "github.com/synapsecns/sanguine/services/sinner/config/server"
+	"github.com/synapsecns/sanguine/services/sinner/db"
+	"github.com/synapsecns/sanguine/services/sinner/db/sql/mysql"
+	"github.com/synapsecns/sanguine/services/sinner/db/sql/sqlite"
+	gqlServer "github.com/synapsecns/sanguine/services/sinner/graphql/server"
+	"golang.org/x/sync/errgroup"
+	"net"
+	"net/http"
+	"os"
+)
+
+var logger = log.Logger("sinner-api")
+
+// Start starts the api server for sinner.
+func Start(ctx context.Context, cfg serverConfig.Config, handler metrics.Handler) error {
+	logger.Warnf("starting api server")
+	router := ginhelper.New(logger)
+	// wrap gin with metrics
+	router.GET(ginhelper.MetricsEndpoint, gin.WrapH(handler.Handler()))
+
+	eventDB, err := InitDB(ctx, cfg.DBFlag, cfg.DBPath, handler, cfg.SkipMigrations)
+	if err != nil {
+		return fmt.Errorf("could not initialize database: %w", err)
+	}
+
+	router.Use(handler.Gin())
+	gqlServer.EnableGraphql(router, eventDB, cfg, handler)
+	if err != nil {
+		return fmt.Errorf("could not create grpc server: %w", err)
+	}
+	fmt.Printf("started graphiql gqlServer on port: http://localhost:%d/graphiql\n", cfg.HTTPPort)
+	g, ctx := errgroup.WithContext(ctx)
+
+	var lc net.ListenConfig
+	listener, err := lc.Listen(ctx, "tcp", fmt.Sprintf(":%d", cfg.HTTPPort))
+	if err != nil {
+		return fmt.Errorf("could not listen on port %d", cfg.HTTPPort)
+	}
+
+	m := cmux.New(listener)
+	httpListener := m.Match(cmux.HTTP1Fast())
```

Comment:
we don't need cmux multiplexing here unless we're also using grpc, just don't wrap the listener